include/xtypes/PrimitiveType.hpp

```diff
@@ -70,6 +70,9 @@ class PrimitiveType : public DynamicType
         : DynamicType(PrimitiveTypeKindTrait<T>::kind, PrimitiveTypeKindTrait<T>::name)
     {}
 
+    PrimitiveType(const PrimitiveType& other) = delete;
+    PrimitiveType(PrimitiveType&& other) = delete;
```

Comment:
Ok with explicitly deleting the move-ctor, shouldn't we delete the move assignment operator as well?
it should be constructed by default if there is no user-declared copy constructor, copy assignment operator or destructor. I might be wrong, but it seems to be the case.

Reply:
Tested. Mark copy constructor to delete invalidates the assignment operator.